lib/phase/flatten.js

```diff
@@ -13,7 +13,7 @@ const LABEL_MATCH = 7;
  * @param  {Object}   newJob    Job to be merged into. For ex: shared settings
  * @param  {Object}   oldJob    Job to merge. For ex: individual job settings
  */
-function merge(newJob, oldJob) {
+function merge(newJob, oldJob, fromTemplate) {
```

Comment:
I think changing the param to be a `config` object or something, since a boolean as the last argument will default to `false` if the function gets called without it. Unless we want `fromTemplate=false` to be the default, I guess.